### DISCUSSION TITLE
Can we make password field optional?

### DISCUSSION THREAD

**Discussion by heyarviind** [pocketbase/pocketbase#1072] [Q&A] [answered]
Hi,

I have a use case where I want to authenticate the user with just email and the OTP, I don't want the user to have a password. 

Is it possible? I have done everything from disabling the email/password login to removing the minimum password length requirement but still it is showing it as a required field.

**Comment by ganigeorgiev** [comment, marked as answer]
No, password is always required.

As a workaround you can set a random password for your users (that's actually what we do when the user is created after OAuth2 sign-up for example).

**Reply by heyarviind** [reply]
Umm yes, this can be done. Thank you for the quick response.

**Reply by dschissler** [reply]
@ganigeorgiev Could you please add the ability to fill in a new auth record with a random password in the admin UI?  This would make it convenient to get up and running creating unverified users that are then sent invitations.  This can proceed creating custom Svelte forms.

Also does the following also sign in the user?

```javascript
await pb.collection('users').confirmVerification('TOKEN');
```

I'm trying to figure out different ways to save time for systems that are open signup and others that are invite only.

**Reply by ganigeorgiev** [reply]
@dschissler I'll consider adding a button to generate a pseudorandom string for a password in the UI.

> Also does the following also sign in the user?
> `await pb.collection('users').confirmVerification('TOKEN');`

No. It only updates the `verified` state of the auth record.

**Reply by dschissler** [reply]
@ganigeorgiev It might be useful to be able to create users with the autogenerated password via the admin UI, and then send them a confirmation email.  Then somehow it demands that they set a password at that time after having automatically logged them in.  I think that the demanding password part would be done completely custom, and probably private fields would be the best situation there because otherwise there is too much risk of having too complicated API rules that could be security issues (somehow).

The reason for the (optional) automatic signup would be to not need to then send a password change request email.  Basically if the two could be combined somehow.

Btw I've very recently played with the signup, signin and emails and it doesn't seem bad to me.  I know that there is some kind of a rework being thoughtup, and I'm not sure if it would be better for me than what it is now.  I like that I can completely control the flow.  Probably it would be hell though if someone wasn't using async await.

**Reply by dschissler** [reply]
@ganigeorgiev 

> I'll consider adding a button to generate a pseudorandom string for a password in the UI.

What about sending meta data to the API endpoint that tells it to scramble the password?  That way you don't have to play with potentially bad Javascript encryption can of worms.  Maybe a `null` value could be sent that would do that (only for admin UI).  In the future perhaps there could be a private auth field called "locked" or something.  I'm not sure, but just giving ideas about how that could go.

I'm thinking that you could provide the means for a lazy developer of a small site to do all sorts of user admin stuff from that interface.  It would be a bit dangerous due to how much is exposed, but that seems ok for something rare on a tiny site.

Then later someone will make a Admin+ package that will just automatically include all of their own Svelte UI to bring it another direction for more specific tools.  In this situation having a bit more built-in always functionality would go a very long way because then they wouldn't have to force database schema changes (or not much) because it would already be there.  It's just a matter of time before someone makes some extra magic routes that bring in huge functionality.
